packages/hardhat-core/test/internal/hardhat-network/provider/modules/eth/methods/sendRawTransaction.ts

```diff
@@ -98,27 +99,56 @@ describe("Eth module", function () {
           assertReceiptMatchesGethOne(receipt, receiptFromGeth, 1);
         });
 
-        describe("Set lower baseFeePerGas", function () {
-          // setting a lower baseFeePerGas here to avoid having to re-create the raw tx
-          useProvider({ initialBaseFeePerGas: 1 });
+        describe("Transaction hash returned within the error data", function () {
+          describe("Set lower baseFeePerGas", function () {
+            // setting a lower baseFeePerGas here to avoid having to re-create the raw tx
+            useProvider({ initialBaseFeePerGas: 1 });
+
+            it("Should return the hash of the failed transaction", async function () {
+              if (!isJsonRpc || isFork) {
+                this.skip();
+              }
+
+              try {
+                // sends a tx with 21000 gas to the 0x1 precompile
+                await this.provider.send("eth_sendRawTransaction", [
+                  "0xf8618001825208940000000000000000000000000000000000000001808082011aa03e2b434ea8994b24017a30d58870e7387a69523b25f153f0d90411a8af8343d6a00c26d36e92d8a8334193b02982ce0b2ec9afc85ad26eaf8c2993ad07d3495f95",
+                ]);
+
+                assert.fail("Tx should have failed");
+              } catch (e) {
+                assert.notInclude(e.message, "Tx should have failed");
+
+                assert.isDefined(e.data.txHash);
+              }
+            });
+          });
+        });
 
-          it("Should return the hash of the failed transaction", async function () {
-            if (!isJsonRpc || isFork) {
-              this.skip();
-            }
+        describe("Base fee validation", function () {
+          // We set an initial base fee too high for the fixture txs
+          useProvider({ initialBaseFeePerGas: 100e9 });
+
+          describe("With automining enabled", function () {
+            it("Should reject txs that can't be mined in the next block", async function () {
+              await assertInvalidInputError(
+                this.provider,
+                "eth_sendRawTransaction",
+                [
+                  "0xf8618001825208940000000000000000000000000000000000000001808082011aa03e2b434ea8994b24017a30d58870e7387a69523b25f153f0d90411a8af8343d6a00c26d36e92d8a8334193b02982ce0b2ec9afc85ad26eaf8c2993ad07d3495f95",
+                ],
+                "too low for the next block, which has a baseFeePerGas of"
+              );
+            });
+          });
 
-            try {
-              // sends a tx with 21000 gas to the 0x1 precompile
+          describe("With automining disabled", function () {
+            it("Should enqueue txs that can't be mined in the next block", async function () {
```

Comment:
Shouldn't we assert something here? At least that the tx exists, to be sure that it's not dropped.

Reply:
yeah, good point.